### PR TITLE
Adding checks for non-modular files not containing modules.yaml

### PR DESCRIPTION
**Pull request by ragabala**
This commit takes care of syncing and publishing non modular content and
checking whether the published content doesn't have modules.yaml. The
following are tested
* Checking whether no modules element is present in the repomd.xml under
repodata
* Checking whether no `modules.yaml.gz` file is present in the published
folder

Closes #4350

### CODE REVIEW

**Comment by bherrin3** [comment]
@kersommoura zhunting and the build team just started the 2.18.1 release. I suggest we wait to merge these PRs until the build is complete to avoid false failures with these changes.

I plan on checking in on the build complete and merging this weekend. 

If you are fine with that, just +1 this and I will merge the PR this weekend.

**Comment by kersommoura** [comment]
@bherrin3 that is not the work flow that we have been following. Tests should be merged without any problem, tests should have code to verify when certain tests should run or be skipped. 
That is why we have piece of code like that.

```python
        if cls.cfg.pulp_version < Version('2.18'):
            raise unittest.SkipTest('This test requires Pulp 2.18 or newer.')
```
All tests tests are verified against the master branch. This helps to find regressions earlier on.

**Comment by bherrin3** [comment]
> All tests tests are verified against the master branch. This helps to find regressions earlier on.

@kersommoura agree. My bad. Forgot about that. 

Will notify @ragabala . Otherwise, everything looked good.

**Comment by kersommoura** [comment]
I will take a look into this PR later today. But good job @ragabala .

**Comment by kersommoura** [comment]
It is passing when running against the 2.18b1 packages.  Please, test against the nightly installation of 2.19.

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py [hunk 1]
```diff
@@ -430,3 +431,71 @@ def test_all(self):
...
+
+        This test does the following
+
+        1. Create and sync a non-modular content.
+        2. Publish the synced content.
+        3. Check whether no modules.yaml is generated for published content.
+
+        This test case targets:
+
+        * `Pulp #4350 <https://pulp.plan.io/issues/4350>`_.
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: Add the issue itself.
* `Pulp #4352 <https://pulp.plan.io/issues/4352>`_.

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py [hunk 2]
```diff
@@ -430,3 +431,71 @@ def test_all(self):
...
+
+
+class CheckModulesYamlTestCase(unittest.TestCase):
+    """Check whether modules.yaml is available if appropriate."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class wide variables."""
+        cls.cfg = config.get_config()
+        if cls.cfg.pulp_version < Version('2.18'):
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modularity.py]:
```suggestion
        if cls.cfg.pulp_version < Version('2.18.1'):
```

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py [hunk 3]
```diff
@@ -430,3 +431,71 @@ def test_all(self):
...
+            raise unittest.SkipTest('This test requires Pulp 2.18 or newer.')
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modularity.py]:
```suggestion
            raise unittest.SkipTest('This test requires Pulp 2.18.1 or newer.')
```

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py [hunk 4]
```diff
@@ -430,3 +431,71 @@ def test_all(self):
...
+            self.cfg,
+            repo['distributors'][0]
+        )
+        self.assertFalse(bool(modules_elements))
+
+    @staticmethod
+    def list_repo_data_files(cfg, repo):
+        """Return a list of all the files present inside repodata dir."""
+        return cli.Client(cfg).run((
+            'find',
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: What about add `-type f ` to assure that just files are returned?

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py [hunk 5]
```diff
@@ -430,3 +431,71 @@ def test_all(self):
...
+    @classmethod
+    def setUpClass(cls):
+        """Create class wide variables."""
+        cls.cfg = config.get_config()
+        if cls.cfg.pulp_version < Version('2.18'):
+            raise unittest.SkipTest('This test requires Pulp 2.18 or newer.')
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+    def test_no_modules_yaml_generated_non_modular(self):
+        """Verify no modules yaml is generated for non modular content.
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modularity.py]:
```suggestion
        """Verify no ``modules.yaml`` is generated for non modular content.
```

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py (hunk 1 above)
Comment (kersommoura) [reply pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: no @ragabala, it was a typo. It was supposed to be ``4252``.

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py (hunk 4 above)
Comment (ragabala) [reply pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: Have Made the changes.

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py (hunk 5 above)
Comment (ragabala) [reply pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: Have made the changes!

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py (hunk 1 above)
Comment (ragabala) [reply pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: Have added the change !

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py (hunk 2 above)
Comment (ragabala) [reply pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: Have added the change in PR #155

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py (hunk 3 above)
Comment (ragabala) [reply pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: Have added the change in PR #155

File: pulp_2_tests/tests/rpm/api_v2/test_modularity.py [hunk 6]
```diff
@@ -430,3 +431,74 @@ def test_all(self):
...
+        return cli.Client(cfg).run((
+            'find',
+            '/var/lib/pulp/published/yum/master/yum_distributor/{}/'.format(
+                repo['id']
+            ),
+            '-type',
+            'f',
+            '-name',
+            '*modules.yaml.gz'
+        )).stdout.splitlines()
```
Comment (dralley) [inline pulp_2_tests/tests/rpm/api_v2/test_modularity.py]: I had to put ```sudo=true``` here to make it work for me.

**Pull request merged**